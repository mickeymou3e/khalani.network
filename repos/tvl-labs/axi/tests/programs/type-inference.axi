function f(x: Nat) -> Boolean;

function g(x: Bytes) -> Boolean;

proc p(x) { f(x) }

forall x. p(x);

forall x, y. (x = y & p(y));

exists x, y. (x = y & p(x));

forall x. ((exists y. y = x) & p(x));

struct Point {
    x: Nat,
    y: Nat,
}

function fp(p: Point) -> Boolean;

forall p. (fp(p) & p.x = 1);

# Type error for g(y).
forall x, y. (x = y & p(x) & g(y));
