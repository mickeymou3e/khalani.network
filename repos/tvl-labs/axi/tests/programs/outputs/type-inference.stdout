Result: Nothing
Result: Nothing
Result: Nothing
Result: Proposition(forall Nat. f[@0:Nat])
Result: Proposition(forall Nat. forall Nat. and(=[@1:Nat, @0:Nat], f[@0:Nat]))
Result: Proposition(exists Nat. exists Nat. and(=[@1:Nat, @0:Nat], f[@1:Nat]))
Result: Proposition(forall Nat. and(exists Nat. =[@0:Nat, @1:Nat], f[@0:Nat]))
Result: Nothing
Result: Nothing
Result: Proposition(forall Point. and(fp[@0:Point], =[Point.x[@0:Point]):Nat, 1]))
